crates/sui/tests/shared_objects_tests.rs

```diff
@@ -400,6 +400,7 @@ async fn shared_object_on_gateway() {
     // to only filter "timeout" errors, but the game way simply returns `anyhow::Error`, this
     // will be fixed by issue #1717. Note that the gateway has an internal retry mechanism but
     // it is not an infinite loop.
+    let mut retry = 10;
```

Comment:
There is no theoretical guarantee that the transaction will go through after 10 retries

Reply:
in theory, (if there are no bugs), you are only guaranteed to have your transaction sequenced "eventually"

Reply:
hmm I have seen this hang in a few cases. What's your recommendation on how we may add some info to help diagnose the hang?

Reply:
I don't think it's new. We have seen it from time to time.

Reply:
This is one example: https://github.com/MystenLabs/sui/runs/6988425031?check_suite_focus=true
I was hoping that this PR will make it easier for you to investigate.

Reply:
10 seems big enough actually. At the time we needed around 2-3 retrials, and most transactions get sequenced at this first attempt